src/System.Buffers.Experimental/System/Buffers/ArraySegmentExtensions.cs

```diff
@@ -2,26 +2,124 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System.Collections.Sequences;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
 
 namespace System.Buffers
 {
     public static class VariousExtensions
```

Comment:
This name is pretty funny. Make it more specific or split up the extension classes